components/calendar/src/islamic.rs

```diff
@@ -218,10 +218,41 @@ impl IslamicTabular {
     }
 }
 
+#[derive(Copy, Clone, Debug, Hash, PartialEq, Eq, PartialOrd, Ord)]
+enum IslamicYearLength {
+    L355,
+    L354,
+    L353,
```

Comment:
nit: document this as an edge case that shouldn't actually occur (and list the positive years/cals for which it does occur)

Reply:
Done